qiskit/aqua/utils/run_circuits.py

```diff
@@ -355,9 +355,19 @@ def run_on_backend(backend, qobj, backend_options=None,
             temp_backend_options = \
                 backend_options['backend_options'] if backend_options != {} else None
             temp_noise_config = noise_config['noise_model'] if noise_config != {} else None
-            job = AerJob(backend, job_id,
-                         backend._run_job, qobj, temp_backend_options, temp_noise_config, False)
-            job._future = job._executor.submit(job._fn, job._job_id, job._qobj, *job._args)
+
+            # Add new options
+            if temp_backend_options is not None or temp_noise_config is not None:
+                config = qobj.config.to_dict()
+                if temp_backend_options is not None:
+                    for key, val in temp_backend_options.items():
+                        config[key] = val if not hasattr(val, 'to_dict') else val.to_dict()
+                if temp_noise_config is not None:
+                    config['noise_model'] = temp_noise_config
+                qobj.config = QasmQobjConfig.from_dict(config)
+
+            job = AerJob(backend, job_id, backend._run, qobj)
+            job.submit()
```

Comment:
This is problematic since it uses a private method and also because this doesn't work on aer < 0.7 (and there is no requirements bump which breaks people using aer 0.6.x). Why not just do

```
job = backend,run(qobj)
job.job_id = job_id
```

Reply:
Changed in #1349